File: frontend/src/modules/membership/presentation/lib/cedula.ts
export const CEDULA_INVALID_MESSAGE = "La cédula debe tener exactamente 10 dígitos.";

export function normalizeCedula(value: string): string {
  return (value || "").replace(/\D/g, "").slice(0, 10);
}

export function isValidEcuadorianCedula(value: string): boolean {
  const raw = (value || "").trim();
  return /^\d{10}$/.test(raw);
}

File: backend/app/modules/membership/domain/cedula.py
CEDULA_INVALID_MESSAGE = "La cédula debe tener exactamente 10 dígitos."


def normalize_cedula(value: str) -> str:
    return "".join(character for character in (value or "") if character.isdigit())


def is_valid_ecuadorian_cedula(value: str) -> bool:
    raw = (value or "").strip()
    return raw.isdigit() and len(raw) == 10
